doc/s3ip_sysfs/s3ip_sysfs_hld.md

```diff
@@ -0,0 +1,339 @@
+
+S3IP sysfs contains two sections:
+
+- sysfs specification: specifies the sysfs directory structure and file content
+- sysfs framework: generate the sysfs directory structure that conforming to the specification.
+
+### 2. S3IP sysfs specification
+#### 2.1 S3IP sysfs specification scenario
+
+Figure2-1 where S3IP sysfs is used.
+
+![scenario](/images/s3ip_sysfs/s3ip_sysfs_scenario.svg)
+
+The S3IP sysfs specification is represented as an organized sysfs directory structure on white-box devices. Device management software and debugging tools need to access hardware through this interface.
+
+Both vendors and users should comply with the S3IP sysfs specification. Device vendors focus on the specification implementation, while users verify the usability of the device against this specification. Vendors should provide software access to every sensor capable of being read. For any available sensor, vendors shall provide S3IP sysfs access.
+
+Figure2-2 S3IP sysfs diretory demo
+
+![demo](/images/s3ip_sysfs/s3ip_sysfs_demo.png)
+
+#### 2.2 Design of the S3ip sysfs specification
+Design Principles
+ - Unified, well-specified behavior, consistent experience among all SONiC devices, regardless of underlying platform; 
+ - Easy to understand, implement, test and debug
+ - The sysfs root directory must be "/sys_switch", and the directory and its subdirectories or files can be soft links. For example, it is legal to link a file from /sys to /sys_switch.
+ - The contents of files in sysfs should be generated dynamically, using the same mechanism as the file contents in the /sys path of the Linux system.
+ - Sysfs has no specific CPU architecture requirements.
+ - Each file in sysfs represents a hardware property, and the property unit should be omitted.
+ - When a switch contains multiple devices of the same type, sysfs must be uniquely identified by the natural numbers 1 to n. For example, /sys_switch/fan/fan1 represents information about fan number 1
+ - This specification defines the path name, permissions, data type, and data unit for each hardware
+ - The sysfs path defined in this specification must exist, and the file content should be "NA" if no hardware is available.
+
+S3IP sysfs specifiction : [specifiction](/doc/s3ip_sysfs/s3ip_sysfs_specification.md)

```

Comment:
spelling error, should be:
S3IP sysfs specification : [specification](/doc/s3ip_sysfs/s3ip_sysfs_specification.md)

Reply:
accepted